pkg/testing/utils/utils.go

```diff
@@ -154,6 +154,40 @@ func BuildProc(comm string, uid, gid uint32) Process {
 	}
 }
 
+// NormalizeParentTid normalizes the parent TID to the first thread (main thread)
+// of the process by reading /proc/$tid/task/ directory. This allows comparing
+// parent TIDs that might be different threads of the same process.
+func NormalizeParentTid(tb testing.TB, p *Process) {
+	tb.Helper()
+	if p.Parent.Tid == 0 {
+		return
+	}
+
+	taskDir := fmt.Sprintf("/proc/%d/task", p.Parent.Tid)
+	entries, err := os.ReadDir(taskDir)
+	if err != nil {
+		// If we can't read the task dir, leave the TID as is
+		tb.Logf("Could not read task dir %s: %v", taskDir, err)
+		return
+	}
+
+	if len(entries) == 0 {
+		tb.Logf("No entries in task dir %s", taskDir)
```

Comment:
This is nice, but I think we can use Errorf(), the execution will continue but the test would be marked as failed.